File: src/main/java/com/company/commitet_jm/entity/TypesFiles.java
package com.company.commitet_jm.entity;

import io.jmix.core.metamodel.datatype.EnumClass;

public enum TypesFiles implements EnumClass<String> {
    EXTERNAL_CODE("A"),
    DATAPROCESSOR("B"),
    REPORT("C"),
    SCHEDULEDJOBS("D"),
    EXCHANGE_RULES("E");

    private final String id;

    TypesFiles(String id) {
        this.id = id;
    }

    @Override
    public String getId() {
        return id;
    }

    public static TypesFiles fromId(String id) {
        for (TypesFiles type : TypesFiles.values()) {
            if (type.id.equals(id)) {
                return type;
            }
        }
        return null;
    }
}

File: src/main/kotlin/com/company/commitet_jm/entity/TypesFiles.kt
package com.company.commitet_jm.entity

import io.jmix.core.metamodel.datatype.EnumClass

enum class TypesFiles(private val id: String) : EnumClass<String> {
    EXTERNAL_CODE("A"),
    DATAPROCESSOR("B"),
    REPORT("C"),
    SCHEDULEDJOBS("D"),
    EXCHANGE_RULES("E");

    override fun getId() = id

    companion object {

        @JvmStatic
        fun fromId(id: String): TypesFiles? = TypesFiles.values().find { it.id == id }
    }
}